src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/services/TestCaseDisplayService.kt

```diff
@@ -122,6 +122,22 @@ class TestCaseDisplayService(private val project: Project) {
             .createProjectScopeChooser(
                 "Insert Test Cases into Class"
             )
+
+        // Warning: The following code is extremely cursed.
+        // It is a workaround for an oversight in the IntelliJ TreeJavaClassChooserDialog.
```

Comment:
When we click the apply button you can choose between search or project tree structure view. The search limits the options only to the classes from the project correct? Not the external libraries, is it possible to look at the implementation of search to see what they do there to restrict the classes or not? Would that make the code less cursed or not?

Reply:
This is what I did to find this workaround. There is an AbstractTreeClassChooserDialog which is inherited by TreeJavaClassChooserDialog (which is the dialog that we instantiate). The constructor of AbstractTreeClassChooserDialog has a parameter to set whether library contents should be shown, but that is hardcoded to true in all constructors of the TreeJavaClassChooserDialog. This is why I use reflection to set it after the object is instantiated.

Reply:
Thank you for the clarifications.